pkg/exasol-rest-api/properties_file_parser.go

```diff
@@ -0,0 +1,45 @@
+package exasol_rest_api
+
+import (
+	"errors"
+	"gopkg.in/yaml.v3"
+	"os"
+)
+
+func getPropertiesFromFile(filepath string, properties *ApplicationProperties) error {
+	propertiesFile, err := openFile(filepath)
+	if err != nil {
+		return err
+	}
+	err = decodePropertiesFile(propertiesFile, properties)
+	if err != nil {
+		return err
+	}
+	closeFile(propertiesFile)
+	return nil
```

Comment:
```suggestion
	propertiesFile, err := openFile(filepath)
	if err != nil {
		return err
	}
	defer closeFile(propertiesFile)
	err = decodePropertiesFile(propertiesFile, properties)
	if err != nil {
		return err
	}
	return nil
```

Watch out. Your code would not close the file in case of an error in `decodePropertiesFile`

Reply:
Good finding, fixed